trader/trader.go

```diff
@@ -180,6 +191,13 @@ func (t *Trader) deleteAllOffers(isAsync bool) {
 			log.Fatalf("%scontinuing to exit after showing error during submission of delete offer ops: %s", logPrefix, e)
 			return
 		}
+
+		t.threadTracker.TriggerGoroutine(func(inputs []interface{}) {
+			e := t.metricsTracker.SendDeleteEvent(false)
+			if e != nil {
+				log.Printf("failed to send update event metric: %s", e)
```

Comment:
TODO NS: comment why we are not throwing an error here

Reply:
Confused - I think this conflicts with [this](https://github.com/stellar/kelp/pull/513#discussion_r500364496)?

Seems like we should at least log the error in `Start`; I'm throwing a `log.Fatalf` because that's the pattern used elsewhere in the file for errors when triggering the goroutine.

Reply:
see above for reasoning about using Printf instead of Fatalf for errors stemming from metrics.

I noticed a bigger issue at line 190 which I didn't catch earlier. When we call `SubmitOps` we pass in a function that is executed asynchronously. This function exits on success (or failure) - line 190. This means that we may never reach the `SendDeleteEvent` call. For this reason alone, we should move this TriggerGoroutine call (line 198) above the call to `SubmitOps` on line 190 (immediately after we enter the `if len(dOps) > 0` case).